packages/integration-sdk-runtime/src/synchronization/batchBySize.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  Entity,
+  IntegrationLogger,
+  Relationship,
+} from '@jupiterone/integration-sdk-core';
+import { UploadDataLookup } from '.';
+import { MAX_BATCH_SIZE } from './shrinkBatchRawData';
+const BATCH_THRESHOLD = 0.9; //will stop chunking if chunk size is between BATCH_THRESHOLD*sizeInBytes and sizeInBytes
+
+export function chunkBySize<T extends UploadDataLookup, K extends keyof T>(
+  data: T[K][],
+  sizeInBytes: number,
+  logger: IntegrationLogger,
+): T[K][][] {
+  return chunk(data, sizeInBytes, logger);
+}
+export function chunk<T extends UploadDataLookup, K extends keyof T>(
```

Comment:
It's a little confusing that there is another function called `chunk` that's used in `synchronization/index.ts` - Also, the functions in this file `chunkBySize` and `chunk` are identical. I recommend just moving your logic into `chunkBySize()` and removing `chunk()` from this file, to avoid making `chunk()` an overloaded function name.